docs/release.md

```diff
@@ -0,0 +1,174 @@
+containing the conformance resources from the FHIR model.
+
+<p align="center">
+<img src="images/npm.png" alt="FHIR NPM" height="200"/>
+</p>
+
+The FHIR packaging specification is still under development and does not have
+a fully fleshed-out release process, but we can still use the versioning,
+package naming, etc. guidelines in the specification as a basis for the
+Kids First DRC release specification.
+
+## Versioning
+We will follow the FHIR NPM package specification's versioning scheme which
+uses [Semantic Versioning](https://semver.org/) to determine version numbers
+for FHIR model releases:
+
+<img src="images/semvar.png" alt="Semantic Versioning Example" height="250"/>
+
+### Examples
+
+1. Create first FHIR model release        -> version `0.1.0`
+2. Create new StructureDefinition(s)      -> version `0.2.0`
+3. Remove a StructureDefinition           -> version `1.0.0`
+4. Change StructureDefinition.id          -> version `2.0.0`
+5. Fix typo in StructureDefinition.title  -> version `2.0.1`
+
+## Release Overview
+
+Kids First DRC FHIR model releases will be managed by Github releases and tags
+on a `release` branch. The NPM package containing the FHIR model will be
+an additional release asset that is part of the Github release.
+
+Here is an overview:
+
+<p align="center">
+<img src="images/release.png" alt="FHIR Model Release"/>
+</p>
+
+
+## Pre-Release
+
+1. Developer creates a feature branch: `add-specimen-conf-resources`
+2. Developer authors the Specimen conformance resources
+   (`StructureDefinition`, `CodeSystem`, `ValueSet`, etc.) and commits locally
+3. Developer runs HL7 FHIR validation on local resources - it passes
+3. Developer pushes changes to remote branch on GitHub
+4. Continuous Integration (CI) pipeline runs:
+    - Unit tests: the same HL7 FHIR validation as in step 3.
+    - Integration test(s): load FHIR model resources (conformance and example)
+      into the integration test FHIR server.
+    - Data migration test(s): if a model change will invalidate data in
+    production, a data migration will be required and must be tested before
+    running in production.
+    - CI pipeline successfully completes.
+5. Developer makes Pull Request for the team to review
+6. Team reviews and approves the Pull Request
+7. Developer merges feature branch into `master` branch
+
+### Testing
+
+The steps in the [Pre-Release](#Pre-Release) include a rigorous testing process
+which includes unit tests and integration tests (read further for details) so
+that there is a very low chance of failure or issues with the FHIR model
+resources themselves when the FHIR model is deployed to the production FHIR
+server.
+
+This helps ensure that clients and users of the FHIR services in production
+will have a high quality experience with minimum bugs in the FHIR model.
+
+## Release Process
+
+## Executed by `fhirmodel release` command
+
+A new CLI command will be added to facilitate the release of the FHIR model
```

Comment:
Do you have a plan for using the same process with other projects? It seems like only steps 2, 3, and 8 are specific to this project. So will this CLI command actually be calling a separate generic release tool and passing it instructions for those steps?

Reply:
Yea I was thinking about this...I think there should be a generic tool that does these things. I didn't explicitly call it out here. Figured that was an implementation detail.

I think that might make the implementation of the release process take longer than if we didn't make it generic but that's probably ok. We can always make releases manually the way we do now until the new release implementation is in place.

Reply:
I believe the kf-release-maker was intended to be the generic Kids First release tool: https://github.com/kids-first/kf-release-maker. We should probably create a new branch to implement this release process. It doesn't have to merge into master for us to use it and test out this new release process on this repo. If things go well, then we can make plans for other projects to migrate to this process.